app/src/main/java/org/p2p/wallet/auth/analytics/GeneralAnalytics.kt

```diff
@@ -2,14 +2,24 @@ package org.p2p.wallet.auth.analytics
 
 import org.p2p.wallet.common.analytics.Analytics
 import org.p2p.wallet.common.analytics.constants.EventNames
+import org.p2p.wallet.common.di.AppScope
+import org.p2p.wallet.sell.interactor.SellInteractor
+import kotlinx.coroutines.launch
 
-class GeneralAnalytics(private val tracker: Analytics) {
+class GeneralAnalytics(
+    private val tracker: Analytics,
+    private val sellInteractor: SellInteractor,
+    private val appScope: AppScope
```

Comment:
god damn, appScope in analytics, what a world we are living in.. 
any ideas on improvement?

Reply:
let's pass parameter in every place it's called